### ISSUE TITLE
False positive ‘useless-super-delegation’ when overriding function signature

### ISSUE THREAD

**Issue by benf-wspdigital** [PyCQA/pylint#1547]
The ‘useless-super-delegation’ check strikes a false positive when the purpose is to override the method's function signature.

An example of this is when providing a different default value for a parameter:

```python
# lorem.py

class LoremIpsum:

    def odio(self, elit="consecteur"):
        print(elit)


class DolorSitAmet(LoremIpsum):

    def odio(self, elit="adipiscing"):
        super().odio(elit)
```

Current behaviour:

```
$ pylint --reports n --enable=useless-super-delegation lorem.py 
************* Module lorem
lorem.py:9: [W0235(useless-super-delegation), DolorSitAmet.odio] Useless super delegation in method 'odio'
```

Expected behaviour:

```
$ pylint --reports n --enable=useless-super-delegation lorem.py 
```

Version:

```
$ pylint --version
pylint 1.7.1, 
astroid 1.5.3
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  5 2015, 21:12:44) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

**Comment by rogalski** [comment]
Thanks for submitting an issue.

It's a duplicate of #1085.
